### PR TITLE
CCIP-3592 Deduplicating plugin e2e test and usdc test

### PR DESCRIPTION
**Pull request by mateusz-sekara**

### CODE REVIEW

File: execute/plugin.go [hunk 1]
```diff
@@ -58,10 +58,6 @@ type Plugin struct {
 	contractsInitialized bool
 }
 
-func (p *Plugin) ReportingCfg() ocr3types.ReportingPluginConfig {
```
Comment (mateusz-sekara) [inline execute/plugin.go:61]: Removing as promised :D

**Review by asoliman92 — APPROVED** [review]

File: internal/libs/testhelpers/common.go [hunk 2]
```diff
@@ -30,3 +32,10 @@ func CreateOracleIDToP2pID(ids ...int) map[commontypes.OracleID]libocrtypes.Peer
 	}
 	return res
 }
+
+func ExtractSequenceNumbers(outcome exectypes.Outcome) []ccipocr3.SeqNum {
+	sequenceNumbers := slicelib.Map(outcome.Report.ChainReports[0].Messages, func(m ccipocr3.Message) ccipocr3.SeqNum {
+		return m.Header.SequenceNumber
+	})
+	return sequenceNumbers
+}
```
Comment (winder) [inline internal/libs/testhelpers/common.go:41]: I don't think this should depend on the execute package. It's only used in one place so maybe leave the original code?

File: execute/test_utils.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+	return msg
+}
+
+func getP2pIDs(ids ...int) map[commontypes.OracleID]libocrtypes.PeerID {
+	res := make(map[commontypes.OracleID]libocrtypes.PeerID)
+	for _, id := range ids {
+		res[commontypes.OracleID(id)] = libocrtypes.PeerID{byte(id)}
+	}
+	return res
+}
```
Comment (winder) [inline execute/test_utils.go:290]: This one could go in `common.go`, `commit/plugin_e2e_test.go` could use it too:
```
	oracleIDToPeerID := make(map[commontypes.OracleID]libocrtypes.PeerID)
	for i := range oracleIDs {
		oracleIDToPeerID[oracleIDs[i]] = peerIDs[i]
	}
```

**Review by winder — APPROVED** [review]
Nice :+1:

File: internal/libs/testhelpers/common.go (hunk 2 above)
Comment (mateusz-sekara) [reply internal/libs/testhelpers/common.go]: Moved to test_utils.go

**Pull request merged**